--- binutils-2.18.50.0.1/bfd/elf64-x86-64.c.orig	2007-09-11 22:49:11.000000000 +1000
+++ binutils-2.18.50.0.1/bfd/elf64-x86-64.c	2007-09-11 22:49:21.000000000 +1000
@@ -965,7 +965,7 @@
 					      symtab_hdr, sym_hashes,
 					      from_type, rel, relend))
     {
-      const reloc_howto_type *from, *to;
+      reloc_howto_type *from, *to;
 
       from = elf64_x86_64_rtype_to_howto (abfd, from_type);
       to = elf64_x86_64_rtype_to_howto (abfd, to_type);
--- binutils-2.18.50.0.1/bfd/elf32-i386.c.orig	2007-09-11 22:45:35.000000000 +1000
+++ binutils-2.18.50.0.1/bfd/elf32-i386.c	2007-09-11 22:45:40.000000000 +1000
@@ -1184,7 +1184,7 @@
 					  symtab_hdr, sym_hashes,
 					  from_type, rel, relend))
     {
-      const reloc_howto_type *from, *to;
+      reloc_howto_type *from, *to;
 
       from = elf_i386_rtype_to_howto (abfd, from_type);
       to = elf_i386_rtype_to_howto (abfd, to_type);
