--- depmod.pl.orig	2006-05-23 22:20:08.000000000 +1000
+++ depmod.pl	2006-05-23 22:23:28.000000000 +1000
@@ -104,7 +104,7 @@
     warn "\nMODULE = $tgtname\n" if $verbose;
 
     # get a list of symbols
-	my @output=`nm $obj`;
+	my @output=`$ENV{CROSS_COMPILE}nm $obj`;
 
     build_ref_tables($tgtname, \@output, $exp, $dep);
 }
@@ -112,7 +112,7 @@
 
 # vmlinux is a special name that is only used to resolve symbols
 my $tgtname = 'vmlinux';
-my @output = $kernelsyms ? `cat $kernelsyms` : `nm $kernel`;
+my @output = $kernelsyms ? `cat $kernelsyms` : `$ENV{CROSS_COMPILE}nm $kernel`;
 warn "\nMODULE = $tgtname\n" if $verbose;
 build_ref_tables($tgtname, \@output, $exp, $dep);
 
