--- gcc-core-4.0.3/gcc/config/t-slibgcc-elf-ver.orig	2006-06-20 01:45:00.000000000 +1000
+++ gcc-core-4.0.3/gcc/config/t-slibgcc-elf-ver	2006-06-20 01:45:17.000000000 +1000
@@ -9,7 +9,7 @@
 SHLIB_OBJS = @shlib_objs@
 SHLIB_DIR = @multilib_dir@
 SHLIB_SLIBDIR_QUAL = @shlib_slibdir_qual@
-SHLIB_LC = -lc
+SHLIB_LC =
 
 SHLIB_LINK = $(GCC_FOR_TARGET) $(LIBGCC2_CFLAGS) -shared -nodefaultlibs \
 	-Wl,--soname=$(SHLIB_SONAME) \
