--- libstdc++-v3/config/io/basic_file_stdio.cc-orig	Sun Mar 21 22:52:12 2004
+++ libstdc++-v3/config/io/basic_file_stdio.cc	Sun Mar 21 22:52:28 2004
@@ -51,9 +51,9 @@
 #endif
 
 // Pick up FIONREAD on Solaris 2.5.
-#ifdef _GLIBCXX_HAVE_SYS_FILIO_H
-#include <sys/filio.h>
-#endif
+//#ifdef _GLIBCXX_HAVE_SYS_FILIO_H
+//#include <sys/filio.h>
+//#endif
 
 #ifdef _GLIBCXX_HAVE_SYS_UIO_H
 #include <sys/uio.h>
