--- libstdc++-v3/config/io/basic_file_stdio.cc.orig	Wed Mar 24 00:02:30 2004
+++ libstdc++-v3/config/io/basic_file_stdio.cc	Wed Mar 24 00:08:48 2004
@@ -41,9 +41,9 @@
 #endif
 
 // Pick up FIONREAD on Solaris 2.5.
-#ifdef _GLIBCPP_HAVE_SYS_FILIO_H
-#include <sys/filio.h>
-#endif
+//#ifdef _GLIBCPP_HAVE_SYS_FILIO_H
+//#include <sys/filio.h>
+//#endif
 
 #ifdef _GLIBCPP_HAVE_POLL
 #include <poll.h>
