--- glibc-2.3.6/sysdeps/unix/sysv/linux/kernel-features.h.orig	2006-05-28 19:44:08.000000000 +1000
+++ glibc-2.3.6/sysdeps/unix/sysv/linux/kernel-features.h	2006-05-28 19:49:44.000000000 +1000
@@ -303,10 +303,13 @@
 # define __ASSUME_MMAP2_SYSCALL		1
 # define __ASSUME_TRUNCATE64_SYSCALL	1
 # define __ASSUME_STAT64_SYSCALL	1
-# define __ASSUME_FCNTL64		1
 # define __ASSUME_VFORK_SYSCALL		1
 #endif
 
+#if __LINUX_KERNEL_VERSION >= 0x020610 && defined __mc68000__
+# define __ASSUME_FCNTL64		1
+#endif
+
 /* Beginning with 2.5.63 support for realtime and monotonic clocks and
    timers based on them is available.  */
 #if __LINUX_KERNEL_VERSION >= 132415
