--- glibc-2.2.5-orig/sysdeps/unix/sysv/linux/m68k/sysdep.h	Fri Apr  1 11:40:12 2005
+++ glibc-2.2.5/sysdeps/unix/sysv/linux/m68k/sysdep.h	Fri Apr  1 11:42:14 2005
@@ -32,6 +32,16 @@
 # define SYS_ify(syscall_name)	__NR_/**/syscall_name
 #endif
 
+/* This is a kludge to make syscalls.list find these under the names
+   pread and pwrite, since some kernel headers define those names
+   and some define the *64 names for the same system calls.  */
+#if !defined __NR_pread && defined __NR_pread64
+# define __NR_pread __NR_pread64
+#endif
+#if !defined __NR_pwrite && defined __NR_pwrite64
+# define __NR_pwrite __NR_pwrite64
+#endif
+
 #ifdef __ASSEMBLER__
 
 /* Linux uses a negative return value to indicate syscall errors, unlike
