diff -pruN kbd-1.12.orig/src/Makefile.in kbd-1.12/src/Makefile.in
--- kbd-1.12.orig/src/Makefile.in	2004-01-04 04:53:54.000000000 +1100
+++ kbd-1.12/src/Makefile.in	2009-10-06 21:30:15.000000000 +1100
@@ -162,7 +162,7 @@ psfxtable: psffontop.o xmalloc.o utf8.o
 loadkeys mapscrn setfont resizecons loadunimap: findfile.o xmalloc.o
 
 chvt clrunimap deallocvt dumpkeys fgconsole getkeycodes getunimap: getfd.o
-kbd_mode loadkeys loadunimap mapscrn resizecons setkeycodes setfont: getfd.o
+kbd_mode loadunimap mapscrn resizecons setkeycodes setfont: getfd.o
 setpalette showconsolefont setlogcons setvesablank showkey: getfd.o
 
 setfont: mapscrn.o loadunimap.o kdmapop.o kdfontop.o psffontop.o utf8.o
diff -pruN kbd-1.12.orig/src/analyze.l kbd-1.12/src/analyze.l
--- kbd-1.12.orig/src/analyze.l	2004-01-17 08:51:44.000000000 +1100
+++ kbd-1.12/src/analyze.l	2009-10-06 21:30:15.000000000 +1100
@@ -1,6 +1,6 @@
 %{
 #include <stdlib.h>
-#include <linux/kd.h>
+#include "kd.h"
 #include "ksyms.h"
 #include "xmalloc.h"
 
@@ -117,7 +117,6 @@ To                      to|To|TO
 .			{return(ERROR); /* report any unknown characters */}
 %%
 #include "ksyms.h"
-#include <linux/keyboard.h>
 
 void
 stringovfl(void) {
diff -pruN kbd-1.12.orig/src/kd.h kbd-1.12/src/kd.h
--- kbd-1.12.orig/src/kd.h	1970-01-01 10:00:00.000000000 +1000
+++ kbd-1.12/src/kd.h	2009-10-06 21:30:15.000000000 +1100
@@ -0,0 +1,59 @@
+#ifndef __LINUX_KD_H__
+#define __LINUX_KD_H__
+#define MAX_NR_KEYMAPS 256
+#define NR_KEYS                128
+#define MAX_NR_FUNC    256
+#define MAX_DIACR      256
+#define KG_SHIFT        0
+#define KG_CTRL         2
+#define KG_ALT          3
+#define KG_ALTGR        1
+#define KG_SHIFTL       4
+#define KG_SHIFTR       5
+#define KG_CTRLL        6
+#define KG_CTRLR        7
+#define KG_CAPSSHIFT    8
+#define K(t,v)          (((t)<<8)|(v))
+#define KTYP(x)                ((x) >> 8)
+#define KVAL(x)                ((x) & 0xff)
+#define K_HOLE          K(KT_SPEC,0)
+#define KT_LATIN        0       /* we depend on this being zero */
+#define KT_LETTER       11      /* symbol that can be acted upon by CapsLock */
+#define KT_FN           1
+#define KT_SPEC         2
+#define KT_PAD          3
+#define KT_DEAD         4
+#define KT_CONS         5
+#define KT_CUR          6
+#define KT_SHIFT        7
+#define KT_META         8
+#define KT_ASCII        9
+#define KT_LOCK         10
+#define KT_SLOCK        12
+#define KDGKBMODE       0x4B44  /* gets current keyboard mode */
+#define KDSKBMODE       0x4B45  /* sets current keyboard mode */
+#define KDGKBENT        0x4B46  /* gets one entry in translation table */
+#define KDSKBENT        0x4B47  /* sets one entry in translation table */
+#define         K_UNICODE       0x03
+#define K_NOSUCHMAP    K(KT_SPEC,127) /* returned by KDGKBENT */
+#define KDSKBSENT       0x4B49  /* sets one function key string entry */
+#define KDSKBDIACR      0x4B4B  /* write kernel accent table */
+
+struct kbentry {
+        unsigned char kb_table;
+        unsigned char kb_index; 
+        unsigned short kb_value;
+};
+struct kbsentry {
+        unsigned char kb_func;
+        unsigned char kb_string[512];
+};
+struct kbdiacr {
+        unsigned char diacr, base, result;
+};
+//struct kbdiacr accent_table[];
+struct kbdiacrs {
+        unsigned int kb_cnt;    /* number of entries in following array */
+        struct kbdiacr kbdiacr[256];    /* MAX_DIACR from keyboard.h */
+};
+#endif
\ No newline at end of file
diff -pruN kbd-1.12.orig/src/ksyms.c kbd-1.12/src/ksyms.c
--- kbd-1.12.orig/src/ksyms.c	2009-10-06 21:29:48.000000000 +1100
+++ kbd-1.12/src/ksyms.c	2009-10-06 21:30:15.000000000 +1100
@@ -1,4 +1,4 @@
-#include <linux/keyboard.h>
+#include "kd.h"
 #include <stdio.h>
 #include <string.h>
 #include "ksyms.h"
diff -pruN kbd-1.12.orig/src/loadkeys.y kbd-1.12/src/loadkeys.y
--- kbd-1.12.orig/src/loadkeys.y	2009-10-06 21:29:48.000000000 +1100
+++ kbd-1.12/src/loadkeys.y	2009-10-06 21:30:46.000000000 +1100
@@ -18,10 +18,8 @@
 #include <fcntl.h>
 #include <ctype.h>
 #include <sys/ioctl.h>
-#include <linux/kd.h>
-#include <linux/keyboard.h>
+#include "kd.h"
 #include "paths.h"
-#include "getfd.h"
 #include "findfile.h"
 #include "modifiers.h"
 #include "nls.h"
@@ -41,7 +39,7 @@ int max_keymap = 0;		/* from here on, de
 int alt_is_meta = 0;
 
 /* the kernel structures we want to set or print */
-u_short *key_map[MAX_NR_KEYMAPS];
+unsigned short *key_map[MAX_NR_KEYMAPS];
 char *func_table[MAX_NR_FUNC];
 struct kbdiacr accent_table[MAX_DIACR];
 unsigned int accent_table_size = 0;
@@ -61,7 +59,7 @@ static void addfunc(struct kbsentry kbs_
 static void killkey(int index, int table);
 static void compose(int diacr, int base, int res);
 static void do_constant(void);
-static void do_constant_key (int, u_short);
+static void do_constant_key (int, unsigned short);
 static void loadkeys(void);
 static void mktable(void);
 static void strings_as_usual(void);
@@ -330,8 +328,6 @@ main(unsigned int argc, char *argv[]) {
 	do_constant();
 	if(optm)
 	        mktable();
-	else
-	        loadkeys();
 	exit(0);
 }
 
@@ -667,7 +663,7 @@ addkey(int index, int table, int keycode
 	if (!defining[table])
 		addmap(table, 0);
 	if (!key_map[table]) {
-	        key_map[table] = (u_short *)xmalloc(NR_KEYS * sizeof(u_short));
+	        key_map[table] = (unsigned short *)xmalloc(NR_KEYS * sizeof(unsigned short));
 		for (i = 0; i < NR_KEYS; i++)
 		  (key_map[table])[i] = K_HOLE;
 	}
@@ -926,7 +922,7 @@ defdiacs(int fd){
 }
 
 void
-do_constant_key (int i, u_short key) {
+do_constant_key (int i, unsigned short key) {
 	int typ, val, j;
 
 	typ = KTYP(key);
@@ -934,7 +930,7 @@ do_constant_key (int i, u_short key) {
 	if ((typ == KT_LATIN || typ == KT_LETTER) &&
 	    ((val >= 'a' && val <= 'z') ||
 	     (val >= 'A' && val <= 'Z'))) {
-		u_short defs[16];
+		unsigned short defs[16];
 		defs[0] = K(KT_LETTER, val);
 		defs[1] = K(KT_LETTER, val ^ 32);
 		defs[2] = defs[0];
@@ -971,7 +967,7 @@ do_constant (void) {
 
 	for (i=0; i<NR_KEYS; i++) {
 		if (key_is_constant[i]) {
-			u_short key;
+			unsigned short key;
 			if (!key_map[r0])
 				lkfatal(_("impossible error in do_constant"));
 			key = (key_map[r0])[i];
@@ -980,27 +976,6 @@ do_constant (void) {
 	}
 }
 
-static void
-loadkeys (void) {
-        int fd;
-        int keyct, funcct, diacct;
-
-	fd = getfd(NULL);
-	keyct = defkeys(fd);
-	funcct = deffuncs(fd);
-	if (accent_table_size > 0 || nocompose)
-		diacct = defdiacs(fd);
-	if (verbose) {
-	        printf(_("\nChanged %d %s and %d %s.\n"),
-		       keyct, (keyct == 1) ? _("key") : _("keys"),
-		       funcct, (funcct == 1) ? _("string") : _("strings"));
-		if (accent_table_size > 0 || nocompose)
-			printf(_("Loaded %d compose %s.\n"), diacct,
-			       (diacct == 1) ? _("definition") : _("definitions"));
-		else
-			printf(_("(No change in compose definitions.)\n"));
-	}
-}
 
 static void strings_as_usual(void) {
 	/*
@@ -1119,7 +1094,7 @@ mktable () {
 	int i, imax, j;
 
 	struct kbsentry kbs;
-	u_char *p;
+	unsigned char *p;
 	int maxfunc;
 	unsigned int keymap_count = 0;
 
@@ -1137,7 +1112,7 @@ mktable () {
 	      keymap_count++;
 	      if (i)
 		   printf("static ");
-	      printf("u_short %s_map[NR_KEYS] = {", mk_mapname(i));
+	      printf("unsigned short %s_map[NR_KEYS] = {", mk_mapname(i));
 	      for (j = 0; j < NR_KEYS; j++) {
 		  if (!(j % 8))
 		    printf("\n");
