diff -aur linux-2.4.27/include/linux/blkdev.h linux-2.4.27-patched/include/linux/blkdev.h
--- linux-2.4.27/include/linux/blkdev.h	Thu Feb 19 00:36:32 2004
+++ linux-2.4.27-patched/include/linux/blkdev.h	Wed Aug 11 13:44:02 2004
@@ -233,7 +233,7 @@
 extern void grok_partitions(struct gendisk *dev, int drive, unsigned minors, long size);
 extern void register_disk(struct gendisk *dev, kdev_t first, unsigned minors, struct block_device_operations *ops, long size);
 extern void generic_make_request(int rw, struct buffer_head * bh);
-extern inline request_queue_t *blk_get_queue(kdev_t dev);
+extern request_queue_t *blk_get_queue(kdev_t dev);
 extern void blkdev_release_request(struct request *);
 
 /*
@@ -246,7 +246,7 @@
 extern void blk_queue_throttle_sectors(request_queue_t *, int);
 extern void blk_queue_make_request(request_queue_t *, make_request_fn *);
 extern void generic_unplug_device(void *);
-extern inline int blk_seg_merge_ok(struct buffer_head *, struct buffer_head *);
+extern int blk_seg_merge_ok(struct buffer_head *, struct buffer_head *);
 
 extern int * blk_size[MAX_BLKDEV];
 
diff -aur linux-2.4.27/include/linux/fs.h linux-2.4.27-patched/include/linux/fs.h
--- linux-2.4.27/include/linux/fs.h	Sun Aug  8 09:26:06 2004
+++ linux-2.4.27-patched/include/linux/fs.h	Wed Aug 11 13:44:00 2004
@@ -1259,7 +1263,7 @@
 }
 
 extern void set_buffer_flushtime(struct buffer_head *);
-extern inline int get_buffer_flushtime(void);
+extern int get_buffer_flushtime(void);
 extern void balance_dirty(void);
 extern int check_disk_change(kdev_t);
 extern int invalidate_inodes(struct super_block *);
diff -aur linux-2.4.27/include/linux/ide.h linux-2.4.27-patched/include/linux/ide.h
--- linux-2.4.27/include/linux/ide.h	Wed Apr 14 23:05:40 2004
+++ linux-2.4.27-patched/include/linux/ide.h	Wed Aug 11 13:45:26 2004
@@ -1445,10 +1445,10 @@
 	void			*special;
 } pkt_task_t;
 
-extern inline void SELECT_DRIVE(ide_drive_t *);
-extern inline void SELECT_INTERRUPT(ide_drive_t *);
-extern inline void SELECT_MASK(ide_drive_t *, int);
-extern inline void QUIRK_LIST(ide_drive_t *);
+extern void SELECT_DRIVE(ide_drive_t *);
+extern void SELECT_INTERRUPT(ide_drive_t *);
+extern void SELECT_MASK(ide_drive_t *, int);
+extern void QUIRK_LIST(ide_drive_t *);
 
 extern void ata_input_data(ide_drive_t *, void *, u32);
 extern void ata_output_data(ide_drive_t *, void *, u32);
diff -aur linux-2.4.27/include/linux/sched.h linux-2.4.27-patched/include/linux/sched.h
--- linux-2.4.27/include/linux/sched.h	Sun Aug  8 09:26:06 2004
+++ linux-2.4.27-patched/include/linux/sched.h	Wed Aug 11 13:44:02 2004
@@ -763,7 +763,7 @@
 extern void end_lazy_tlb(struct mm_struct *mm);
 
 /* mmdrop drops the mm and the page tables */
-extern inline void FASTCALL(__mmdrop(struct mm_struct *));
+extern void FASTCALL(__mmdrop(struct mm_struct *));
 static inline void mmdrop(struct mm_struct * mm)
 {
 	if (atomic_dec_and_test(&mm->mm_count))
diff -aur linux-2.4.27/net/appletalk/ddp.c linux-2.4.27-patched/net/appletalk/ddp.c
--- linux-2.4.27/net/appletalk/ddp.c	Tue Sep 11 00:57:00 2001
+++ linux-2.4.27-patched/net/appletalk/ddp.c	Wed Aug 11 13:53:46 2004
@@ -106,8 +106,8 @@
 #endif /* APPLETALK_DEBUG */
 
 #ifdef CONFIG_SYSCTL
-extern inline void atalk_register_sysctl(void);
-extern inline void atalk_unregister_sysctl(void);
+extern void atalk_register_sysctl(void);
+extern void atalk_unregister_sysctl(void);
 #endif /* CONFIG_SYSCTL */
 
 struct datalink_proto *ddp_dl, *aarp_dl;
--- linux-2.4.27/include/linux/reiserfs_fs.h	Fri Aug 13 02:04:13 2004
+++ linux-2.4.27-patched/include/linux/reiserfs_fs.h	Fri Aug 13 02:11:00 2004
@@ -1771,15 +1771,15 @@
 /* stree.c */
 int B_IS_IN_TREE(const struct buffer_head *);
 extern inline void copy_short_key (void * to, const void * from);
-extern inline void copy_item_head(struct item_head * p_v_to, 
+extern void copy_item_head(struct item_head * p_v_to, 
 								  const struct item_head * p_v_from);
 
 // first key is in cpu form, second - le
 extern inline int comp_keys (const struct key * le_key, 
 			     const struct cpu_key * cpu_key);
-extern inline int  comp_short_keys (const struct key * le_key, 
+extern int  comp_short_keys (const struct key * le_key, 
 				    const struct cpu_key * cpu_key);
-extern inline void le_key2cpu_key (struct cpu_key * to, const struct key * from);
+extern void le_key2cpu_key (struct cpu_key * to, const struct key * from);
 
 // both are cpu keys
 extern inline int comp_cpu_keys (const struct cpu_key *, const struct cpu_key *);
@@ -1788,8 +1788,8 @@
 extern inline void cpu_key2cpu_key (struct cpu_key *, const struct cpu_key *);
 
 // both are in le form
-extern inline int comp_le_keys (const struct key *, const struct key *);
-extern inline int comp_short_le_keys (const struct key *, const struct key *);
+extern int comp_le_keys (const struct key *, const struct key *);
+extern int comp_short_le_keys (const struct key *, const struct key *);
 
 //
 // get key version from on disk key - kludge
@@ -1824,7 +1824,7 @@
 int search_for_position_by_key (struct super_block * p_s_sb, 
 								const struct cpu_key * p_s_cpu_key, 
 								struct path * p_s_search_path);
-extern inline void decrement_bcount (struct buffer_head * p_s_bh);
+extern void decrement_bcount (struct buffer_head * p_s_bh);
 void decrement_counters_in_path (struct path * p_s_search_path);
 void pathrelse (struct path * p_s_search_path);
 int reiserfs_check_path(struct path *p) ;
@@ -1902,7 +1902,7 @@
 void i_attrs_to_sd_attrs( struct inode *inode, __u16 *sd_attrs );
 
 /* namei.c */
-inline void set_de_name_and_namelen (struct reiserfs_dir_entry * de);
+void set_de_name_and_namelen (struct reiserfs_dir_entry * de);
 int search_by_entry_key (struct super_block * sb, const struct cpu_key * key, 
 			 struct path * path, 
 			 struct reiserfs_dir_entry * de);
@@ -2050,7 +2050,7 @@
                       struct buffer_head **);
 
 /* do_balance.c */
-inline void do_balance_mark_leaf_dirty (struct tree_balance * tb, 
+void do_balance_mark_leaf_dirty (struct tree_balance * tb, 
 					struct buffer_head * bh, int flag);
 #define do_balance_mark_internal_dirty do_balance_mark_leaf_dirty
 #define do_balance_mark_sb_dirty do_balance_mark_leaf_dirty
--- linux-2.4.27/drivers/video/riva/fbdev.c	Fri Aug 13 01:53:13 2004
+++ linux-2.4.27-patched/drivers/video/riva/fbdev.c	Fri Aug 13 01:53:28 2004
@@ -117,7 +117,7 @@
 static void rivafb_blank(int blank, struct fb_info *info);
 
 extern void riva_setup_accel(struct rivafb_info *rinfo);
-extern inline void wait_for_idle(struct rivafb_info *rinfo);
+extern void wait_for_idle(struct rivafb_info *rinfo);
 
 
 
--- linux-2.4.27/include/net/ip.h	Fri Aug 13 02:14:43 2004
+++ linux-2.4.27-patched/include/net/ip.h	Fri Aug 13 02:15:10 2004
@@ -137,7 +137,7 @@
 void ip_send_reply(struct sock *sk, struct sk_buff *skb, struct ip_reply_arg *arg,
 		   unsigned int len); 
 
-extern __inline__ int ip_finish_output(struct sk_buff *skb);
+extern int ip_finish_output(struct sk_buff *skb);
 
 struct ipv4_config
 {
--- linux-2.4.27/fs/reiserfs/Makefile-orig	Mon Aug 25 21:44:43 2003
+++ linux-2.4.27/fs/reiserfs/Makefile	Sat Aug 14 02:07:43 2004
@@ -21,6 +21,11 @@
 # add it here.
 ifeq ($(CONFIG_PPC32),y)
 EXTRA_CFLAGS := -O1
+else
+# gcc 3.4.1 doesn't inline strcpy with -O2!
+ifeq ($(CONFIG_X86),y)
+EXTRA_CFLAGS := -O1
+endif
 endif
 
 include $(TOPDIR)/Rules.make
--- linux-2.4.27/drivers/sound/Makefile-orig	Sat Aug 14 02:34:23 2004
+++ linux-2.4.27/drivers/sound/Makefile	Sat Aug 14 02:40:27 2004
@@ -3,6 +3,10 @@
 # 18 Apr 1998, Michael Elizabeth Chastain, <mailto:mec@shout.net>
 # Rewritten to use lists instead of if-statements.
 
+# gcc 3.4.1 doesn't inline strcpy with -O2!
+ifeq ($(CONFIG_X86),y)
+EXTRA_CFLAGS := -O1
+endif
 
 # All of the (potential) objects that export symbols.
 # This list comes from 'grep -l EXPORT_SYMBOL *.[hc]'.
--- linux-2.4.27/arch/alpha/kernel/pci_iommu.c-orig	Mon Aug 16 02:57:12 2004
+++ linux-2.4.27/arch/alpha/kernel/pci_iommu.c	Mon Aug 16 02:57:28 2004
@@ -503,7 +503,7 @@
 /* Given a scatterlist leader, choose an allocation method and fill
    in the blanks.  */
 
-static inline int
+static int
 sg_fill(struct scatterlist *leader, struct scatterlist *end,
 	struct scatterlist *out, struct pci_iommu_arena *arena,
 	dma_addr_t max_dma, int dac_allowed)
