--- a/Makefile	2008-01-25 09:58:37.000000000 +1100
+++ b/Makefile	2008-02-10 01:19:18.000000000 +1100
@@ -308,7 +308,7 @@
 OBJDUMP		= $(CROSS_COMPILE)objdump
 AWK		= awk
 GENKSYMS	= scripts/genksyms/genksyms
-DEPMOD		= /sbin/depmod
+DEPMOD		= depmod.pl
 KALLSYMS	= scripts/kallsyms
 PERL		= perl
 CHECK		= sparse
@@ -1038,11 +1038,6 @@
 
 PHONY += _modinst_
 _modinst_:
-	@if [ -z "`$(DEPMOD) -V 2>/dev/null | grep module-init-tools`" ]; then \
-		echo "Warning: you may need to install module-init-tools"; \
-		echo "See http://www.codemonkey.org.uk/docs/post-halloween-2.6.txt";\
-		sleep 1; \
-	fi
 	@rm -rf $(MODLIB)/kernel
 	@rm -f $(MODLIB)/source
 	@mkdir -p $(MODLIB)/kernel
@@ -1525,8 +1520,6 @@
       cmd_depmod = \
-	if [ -r System.map -a -x $(DEPMOD) ]; then                              \
-		$(DEPMOD) -ae -F System.map                                     \
-		$(if $(strip $(INSTALL_MOD_PATH)), -b $(INSTALL_MOD_PATH) -r)   \
-		$(KERNELRELEASE);                                               \
+	if [ -r System.map ]; then                              		\
+		NM=$(CROSS_COMPILE)nm $(DEPMOD) -a -F System.map -b $(INSTALL_MOD_PATH)/lib/modules/$(KERNELRELEASE); \
 	fi
 
 # Create temporary dir for module support files
