diff -ru a/arch/powerpc/Makefile b/arch/powerpc/Makefile
--- a/arch/powerpc/Makefile	2007-02-21 17:08:51.000000000 +1100
+++ b/arch/powerpc/Makefile	2007-02-21 17:09:24.000000000 +1100
@@ -35,7 +35,7 @@
 
 export CROSS32CC CROSS32AS CROSS32LD CROSS32AR CROSS32OBJCOPY
 
-KBUILD_DEFCONFIG := $(shell uname -m)_defconfig
+KBUILD_DEFCONFIG := pmac32_defconfig
 
 ifeq ($(CONFIG_PPC64),y)
 OLDARCH	:= ppc64
